File: piper/android/app/src/main/java/io/gopiper/piper/engine/statements/controls_if.kt
package io.gopiper.piper.engine.statements

import android.util.Log
import io.gopiper.piper.engine.Piper
import org.json.JSONObject
import java.lang.Exception

suspend fun Piper.executeControlsIf(statement: JSONObject, onError: (e: Exception) -> Unit) {
    try {
        Log.d("hello", "execute_controls_if")
        val pointer = statement.getJSONArray("pointer")
        for (i in 0 until pointer.length()) {
            val block = pointer.getJSONObject(i)
            if (block.has("else")) {
                execute(block.getString("else"))
            } else {
                val cond = executeStatement(block.get("cond").toString())
                if (cond is Boolean) {
                    if (cond) {
                        val doBlock = block.get("do").toString().trim()
                        if (doBlock != "") execute(doBlock)
                        break
                    }
                }
            }
        }
    } catch (e: Exception) {
        onError(e)
    }
}

File: piper/extensions/chrome/src/engine/statements/controls_if.js
import { stringify } from "../util"


export default async function(statement, onError) {
    try {
        for (let block of statement.pointer) {
            if ("else" in block) {
                await this.execute(stringify(block.else))
            } else {
                let cond = await this.executeStatement(stringify(block.cond))
                if (typeof cond === "boolean" && cond) {
                    let doBlock = stringify(block.do).trim()
                    if (doBlock !== "") await this.execute(doBlock)
                    break
                }
            }
        }
    } catch (e) {
        onError(e)
    }
}
